app/controllers/api/functions.php

```diff
@@ -845,6 +845,93 @@
         ]), Response::MODEL_DEPLOYMENT_LIST);
     });
 
+App::get('/v1/functions/:functionId/deployments/:deploymentId/download')
+    ->groups(['api', 'functions'])
+    ->desc('Download Deployment')
+    ->label('scope', 'functions.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'downloadDeployment')
+    ->label('sdk.description', '/docs/references/functions/download-deployment.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', '*/*')
+    ->label('sdk.methodType', 'location')
+    ->param('functionId', '', new UID(), 'Function ID.')
+    ->param('deploymentId', '', new UID(), 'Deployment ID.')
+    ->inject('response')
+    ->inject('request')
+    ->inject('dbForProject')
+    ->inject('deviceFunctions')
+    ->action(function (string $functionId, string $deploymentId, Response $response, Request $request, Database $dbForProject, Device $deviceFunctions) {
```

Comment:
Do we also read this file in chunks from the imports worker? It is important to make sure we don't load the entire file to memory during migration.